mapper-runtime/src/main/java/com/datastax/oss/driver/api/mapper/annotations/WithStatementAttributes.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.mapper.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.CLASS)
+public @interface WithStatementAttributes {
+
+  String executionProfileName() default "";
+
+  int pageSize() default Integer.MIN_VALUE;
+
+  boolean idempotence() default false;
+
+  String consistencyLevel() default "";
```

Comment:
Why we are not using `com.datastax.oss.driver.api.core.ConsistencyLevel` here instead of a `String`?

Reply:
Annotations don't support complex objects as data types.